src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -508,46 +526,74 @@ public static void CopyTo_Invalid()
             {
                 if (comparisonType == StringComparison.CurrentCulture)
                 {
                     // Use Compare(string, string) or Compare(string, string, false) or CompareTo(string)
                     Assert.Equal(expected, Math.Sign(string.Compare(strA, strB)));
-                    Assert.Equal(expected, Math.Sign(string.Compare(strA, strB, false)));
+                    Assert.Equal(expected, Math.Sign(string.Compare(strA, strB, ignoreCase: false)));
                     if (strA != null)
                     {
                         Assert.Equal(expected, Math.Sign(strA.CompareTo(strB)));
 
                         IComparable iComparable = strA;
                         Assert.Equal(expected, Math.Sign(iComparable.CompareTo(strB)));
                     }
+                    if (strB != null)
+                    {
+                        Assert.Equal(expected, -Math.Sign(strB.CompareTo(strA)));
+
+                        IComparable iComparable = strB;
+                        Assert.Equal(expected, -Math.Sign(iComparable.CompareTo(strA)));
+                    }
                 }
                 else if (comparisonType == StringComparison.CurrentCultureIgnoreCase)
                 {
                     // Use Compare(string, string, true)
-                    Assert.Equal(expected, Math.Sign(string.Compare(strA, strB, true)));
+                    Assert.Equal(expected, Math.Sign(string.Compare(strA, strB, ignoreCase: true)));
                 }
                 else if (comparisonType == StringComparison.Ordinal)
                 {
                     // Use CompareOrdinal(string, string)
                     Assert.Equal(expected, Math.Sign(string.CompareOrdinal(strA, strB)));
                 }
-                // Use CompareOrdinal(string, string, StringComparisonType)
+                // Use CompareOrdinal(string, string, StringComparison)
                 Assert.Equal(expected, Math.Sign(string.Compare(strA, strB, comparisonType)));
             }
             if (comparisonType == StringComparison.CurrentCulture)
             {
-                // Use Compare(string, int, string, int, int)
-                Assert.Equal(expected, Math.Sign(string.Compare(strA, indexA, strB, indexB, length)));
+                // This may have different behavior than the overload accepting a StringComparison
+                // for a combination of null/invalid inputs; see notes in Compare_Invalid for more
+
+                if (!skipNonComparisonOverloads)
+                {
+                    // Use Compare(string, int, string, int, int) or Compare(string, int, string, int, int, false)
+                    Assert.Equal(expected, Math.Sign(string.Compare(strA, indexA, strB, indexB, length)));
+                    // Uncomment when this is exposed in .NET Core
+                    // Assert.Equal(expected, Math.Sign(string.Compare(strA, indexA, strB, indexB, length, ignoreCase: false)));
```

Comment:
Is there an issue for adding back the relevant methods on string?  If yes, we should include a TODO and the issue number in the "Uncomment" comment.  If no, we should open one and then do it.  And if these have already been added in the dev/api branch, it would be better to open this PR against that branch so that these can be uncommented there.